starter-core/src/main/java/io/micronaut/starter/feature/messaging/kafka/Kafka.java

```diff
@@ -56,10 +69,10 @@ public String getDescription() {
 
     @Override
     public void apply(GeneratorContext generatorContext) {
-        generatorContext.addDependency(Dependency.builder()
-                .groupId("io.micronaut.kafka")
-                .artifactId("micronaut-kafka")
-                .compile());
+        generatorContext.addDependency(MICRONAUT_KAFKA);
+        if (generatorContext.hasFeature(TestContainers.class)) {
```

Comment:
Please, move this logic to the TestContainers feature. If Kafka feature is present add an additional dependency.